src/commands/login.ts

```diff
@@ -94,80 +128,33 @@ export const handler = async (args: yargs.Arguments): Promise<any> => {
-                SimbaConfig.log.debug(`:: EXIT :`);
-                return;
-            }
-            if (!org || !app) {
-                const orgFromSimbaJson = SimbaConfig.ProjectConfigStore.get("organisation");
-                const orgName = orgFromSimbaJson.name;
-                if (!orgName) {
-                    SimbaConfig.log.error(`${chalk.redBright(`no organisation specified in your login command, and no organisation present in your simba.json. Please login in non-inetractive mode and choose your organisation, or use the --org <org> flag in your non-interactive login command.`)}`);
-                    return;
-                } else {
-                    SimbaConfig.log.info(`${chalk.cyanBright(`no org was specified in login command; logging in using org ${orgName} from simba.json`)}`);
-                }
-                const appFromSimbaJson = SimbaConfig.ProjectConfigStore.get("application");
-                const appName = appFromSimbaJson.name;
-                if (!appName) {
-                    SimbaConfig.log.error(`${chalk.redBright(`no app specified in your login command, and no application present in your simba.json. Please login in non-inetractive mode and choose your application, or use the --app <app> flag in your non-interactive login command.`)}`);
-                    return;
-                } else {
-                    SimbaConfig.log.info(`${chalk.cyanBright(`no app was specified in login command; logging in using app ${appName} from simba.json`)}`);
-                }
-            }
-            authStore.logout();
+    try {
+        authStore.logout();
+        if (!authStore.isLoggedIn()) {
+            await authStore.performLogin();
+        } else {
             try {
-                await authStore.performLogin(interactive);
-                if (org) {
-                    await chooseOrganisationFromName(simbaConfig, org as string);
-                }
-                if (app) {
-                    await chooseApplicationFromName(simbaConfig, app as string);
-                }
-                SimbaConfig.resetSimbaJson(previousSimbaJson, org);
-                SimbaConfig.log.info(`${chalk.greenBright(`Logged in to SIMBA Chain!`)}`);
-                SimbaConfig.log.debug(`:: EXIT :`);
-                return;
-            } catch (error) {
-                if (axios.isAxiosError(error) && error.response) {
-                    SimbaConfig.log.error(`${chalk.redBright(`\nsimba: EXIT : ${JSON.stringify(error.response.data)}`)}`)
-                } else {
-                    SimbaConfig.log.error(`${chalk.redBright(`\nsimba: EXIT : ${JSON.stringify(error)}`)}`);
-                }
-                return;
-            }
-        }
-        try {
-            authStore.logout();
-            if (!authStore.isLoggedIn()) {
+                await authStore.refreshToken();
```

Comment:
so the isLoggedIn() functionality works differently for the azureHandler and the KeycloakHandler. the KeycloakHandler's isLoggedIn() functionality checks to see if we've started the process of acquiring a token for device login. The AzureHandler's isLoggedIn() functionality checks to see if a server is going I think. I'm not sure if running refreshToken() would cause any issues in that flow.